### PR TITLE
Not Diamond Extension

### PR DESCRIPTION
**Pull request by dragos199993**
Hi there! I am a big fan of Raycast and would love to integrate Not Diamond into it. In case you’re unfamiliar with Not Diamond, it automatically routes your prompts to the AI model best-suited to respond to that prompt. By using Not Diamond as an AI “meta-model,” you can achieve state-of-the-art performance that beats every foundation model on every major benchmark.*

For adding Not Diamond to Raycast we would like to add an extension with the following features:

- [x] Add Not Diamond API Key and optional other provider API Keys + change the Tradeoff value
- [x] After initial configuration, based on the added providers, the user can select from a multi select list all available providers supported by Not Diamond
- [x] Simple chat with history. For now there is no history conversation, but based on feedback we can add it on a later stage
- [x] After getting a result, copy to clipboard to ease the day by day workflow

We didn't tried to battle how Raycast is built and we used all of the outbox UI components and functionality in order to not deviate from the original idea.

**If you’d like to learn more about our routing, check out [[this example walkthrough video with Tomas](https://x.com/tomas_hk/status/1818296673749328199)](https://x.com/tomas_hk/status/1818296673749328199), our CEO. You can also* 

- *see how we work by using [[our Chat app](https://chat.notdiamond.ai/)](https://chat.notdiamond.ai/),*
- *learn more at [[our website](https://www.notdiamond.ai/)](https://www.notdiamond.ai), or*
- *[[read our docs](https://notdiamond.readme.io/)](https://notdiamond.readme.io/).*

### CODE REVIEW

**Comment by dragos199993** [comment]
Hello 👋! I am currently facing a problem with the maintainer rights pipeline. Following the link from the error, there is no such checkbox there. Am I missing something 🤔? Thanks for the help! 🙏

File: extensions/not-diamond/CHANGELOG.md
Comment (pernielsentikaer) [inline extensions/not-diamond/CHANGELOG.md:3]: Please set the current data if you can't change the modify rights for our bot 🙂

File: extensions/not-diamond/package.json
Comment (pernielsentikaer) [inline extensions/not-diamond/package.json:17]:
```suggestion
      "subtitle": "",
```
Should not be used to describe the command

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- Could you please compress metadata images yourself if you can't change the modify rights for our bot 🙂
- The image `metadata/4.png` doesn’t bring any contextual value, can we remove that?
- Missing icon in the action panel

not-diamond 2024-10-29 at 09 31 52

- Maybe switch actions?

not-diamond 2024-10-29 at 09 32 24

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/not-diamond/package.json
Comment (dragos199993) [reply extensions/not-diamond/package.json]: hmm there seems to be a validation on the length for subtitle. This will throw an error on pipeline 
```
Error:   17:18  error  length must be >= 2, but got 0
Error:   17:18  error  does not match pattern '^[^\\s]+(?: [^\\s]+)*$'
```

any advice on what should I add on this one just to pass the checks? 🙏 thanks!

File: extensions/not-diamond/CHANGELOG.md
Comment (dragos199993) [reply extensions/not-diamond/CHANGELOG.md]: This is done ✅

**Comment by dragos199993** [comment]
hi @pernielsentikaer! glad to meet you! I addressed most of the feedback except the last point `Maybe switch actions?` and the subtitle being empty
Contacted you also in Slack as I am not sure what is expected there.

Thanks for help 🙏 !

File: extensions/not-diamond/package.json
Comment (dragos199993) [reply extensions/not-diamond/package.json]: As advised, removed the line fixed this ✅

Comment (pernielsentikaer) [reply extensions/not-diamond/package.json]: Yeah, sorry about that 🙂

**Comment by pernielsentikaer** [comment]
Is this ready, or do you have more things coming?

**Comment by dragos199993** [comment]
> Is this ready, or do you have more things coming?

hey @pernielsentikaer this is ready. Apologies, I was waiting for the review from you 😅  didn't knew you were expecting a confirmation. All good now

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**
